utils/prompt_client_cli.py

```diff
@@ -0,0 +1,430 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# SPDX-FileCopyrightText: © 2024 Tenstorrent AI ULC
+
+import os
+import getpass
+import threading
+import logging
+import json
+import argparse
+import time
+from datetime import datetime
+import requests
+from pathlib import Path
+from concurrent.futures import ThreadPoolExecutor, as_completed
+
+import jwt
+import numpy as np
+from transformers import AutoTokenizer
+
+from utils.prompt_generation import add_prompt_gen_args, generate_prompts
+
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+)
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+# set numpy seed for reproducibility
+np.random.seed(42)
+
+
+def get_authorization():
+    authorization = os.getenv("AUTHORIZATION", None)
+    if authorization is None:
+        jwt_secret = os.getenv("JWT_SECRET", None)
+        if jwt_secret is None:
+            raise ValueError(
+                "Neither AUTHORIZATION or JWT_SECRET environment variables are set."
+            )
+        json_payload = json.loads('{"team_id": "tenstorrent", "token_id":"debug-test"}')
+        encoded_jwt = jwt.encode(json_payload, jwt_secret, algorithm="HS256")
+        authorization = f"{encoded_jwt}"
+    return authorization
+
+
+def get_api_base_url():
+    DEPLOY_URL = os.getenv("DEPLOY_URL", "http://127.0.0.1")
+    base_url = f"{DEPLOY_URL}:{os.getenv('SERVICE_PORT', '8000')}/v1"
+    return base_url
+
+
+def get_api_url():
+    base_url = get_api_base_url()
+    api_url = f"{base_url}/completions"
+    return api_url
+
+
+# Thread-safe data collection
+responses_lock = threading.Lock()
+responses = []
+
+
+def call_inference_api(
+    prompt,
+    response_idx,
+    prompt_len,
+    stream,
+    headers,
+    api_url,
+    max_tokens,
+    vll_model,
```

Comment:
should this be vllm_model ?

Reply:
yes, that's the model lookup vLLM is hosting it on.